README.md

```diff
@@ -1 +1,219 @@
-# chat-firebase-swift
\ No newline at end of file
+# QLFirebaseChat
+
+QLFirebaseChat is a Swift package that simplifies chat functionality using Firebase Firestore. This package provides an easy way to integrate real-time chat features into your Swift applications.
+
+## Installation
+
+You can install QLFirebaseChat using Swift Package Manager (SPM). Follow these steps to add it to your Xcode project:
+
+ Add the library as a Swift package dependency.
+
+- In Xcode, open your project.
+- Go to "File" > "Swift Packages" > "Add Package Dependency..."
+- Enter the following URL: `https://github.com/Quokka-Labs-LLP/chat-firebase-swift.git`
+- Select the version or branch you` want to use.
+- Click "Finish."
+- Import package with `import QLFirebaseChat`
+
+ ## Usage
+
+ ### Initialize Firebase
+
+To use Firebase services in your chat application, you need to initialize Firebase. Call the `initialize` method to configure Firebase. You should typically do this early in your app's lifecycle, such as in your app delegate's `application(_:didFinishLaunchingWithOptions:)` method.
+
+```swift
+QLChatFirebase.initialize()
+```
+This method configures Firebase with the settings specified in your project's Firebase configuration file.
+
+Make sure you have added your Firebase configuration file (usually named GoogleService-Info.plist) to your Xcode project.
+
+```swift
+import QLChatFirebase
+
+// Initialize Firebase in your app delegate
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    // Initialize Firebase for your chat functionality
+    // Option 1: Using QLChatFirebase
+    QLChatFirebase.initialize()
+
+    // Option 2: Using Firebase directly
+     FirebaseApp.initialize()
+    return true
+}
+```
+Once Firebase is initialized, you can use other Firebase services, such as Firestore and Authentication, in your chat application.
+
+Note: Ensure you have set up Firebase in your project by following the Firebase setup instructions provided by Firebase.
+
+ ## Usage of Firebase Authentication:
+ - Initialize an instance of FirebaseAuthManager\
+   `let authManager = FirebaseAuthManager.shared'
+
+ ### Register a New User
+
+To register a new user, use the `registerUser` method. Provide the user's email and password as parameters. Here's an example:
+
+```swift
+authManager.registerUser(email: "user@example.com", password: "password") { result in
```

Comment:
**suggestion**: class and method names must be personalized to the module, even if they just relay the control to firebase. That way, you'd have more freedom to change the implementations within without changing the names.

Names shall be important to finalize as we can push developers to change class and method names everytime they download an update.

Reply:
This applies to all classes and methods. Classes, in any way, should be 'QL' as prefix.

Reply:
This can be framed as user registering himself to use the chat. We should let know if developers are using firebase auth - they don't have to use this method.